local/bin/py/build/content_manager.py

```diff
@@ -209,14 +220,62 @@ def prepare_content(configuration, github_token, extract_dir):
     :param extract_dir: Directory into which to put all content downloaded.
     """
     try:
-        list_of_contents = local_or_upstream(
-            github_token, extract_dir, extract_config(configuration))
+        extract_sourced_and_cached_content_from_pull_config(self, configuration)
+        fetch_sourced_content_from_local_or_upstream(self, github_token, extract_dir)
     except Exception as e:
         if not getenv("CI_COMMIT_REF_NAME"):
-            print(
-                f"\x1b[33mWARNING\x1b[0m: Downloading files failed, documentation is now in degraded mode. {e}")
+            print(f"\x1b[33mWARNING\x1b[0m: Downloading files failed, documentation is now in degraded mode. {e}")
         else:
-            print(
-                "\x1b[31mERROR\x1b[0m: Downloading files failed, stopping build.")
+            print("\x1b[31mERROR\x1b[0m: Downloading files failed, stopping build.")
             sys.exit(1)
-    return list_of_contents
+
+
+def download_and_extract_cached_files_from_s3():
+    s3_url = f'https://origin-static-assets.s3.amazonaws.com/build_artifacts/master/latest-cached.tar.gz'
```

Comment:
how does this file get created? what happens if it isn't there?

Reply:
@nsollecito the [cached artifact gets created after each master deploy](https://github.com/DataDog/documentation/blob/master/.gitlab-ci.yml#L264).   the tarred `latest-cached` file in s3 gets overwritten with the most recent files from the master build each time, and this has been running in prod for a little bit now.

while the `latest-cached` file should always exist, there is a [mechanism in place to retry the full build](https://github.com/DataDog/documentation/blob/brian.deutsch/cache-single-sourced-content/local/bin/py/build/update_pre_build.py#L124-L136) (without caching) if downloading files from cache should fail.